python_modules/dagster/dagster/_core/execution/retries.py

```diff
@@ -115,11 +113,16 @@ def auto_reexecution_should_retry_run(instance: "DagsterInstance", run: DagsterR
     We think this is an acceptable tradeoff to make since the automatic reexecution system won't launch more than max_retries
     run itself, just that max_retries + 1 runs could be launched in total if a manual retry is timed to cause this condition (unlikely).
     """
-    from dagster._core.events import RunFailureReason
-
     if run.status != DagsterRunStatus.FAILURE:
         return False
 
+    retry_on_asset_or_op_failure = get_boolean_tag_value(
```

Comment:
realized i can move this block up in the fn and get rid of some nesting of if clauses